cdap-app-fabric/src/main/java/co/cask/cdap/metadata/LineageHandler.java

```diff
@@ -164,28 +165,21 @@ public void datasetFieldLineageDetails(HttpRequest request, HttpResponder respon
     List<FieldOperationInfo> fieldOperationInfos = new ArrayList<>();
     EndPoint sourceEndPoint = EndPoint.of("default", "file");
     EndPoint targetEndPoint = EndPoint.of("default", "another_file");
-    FieldOperationInfo operationInfo = new FieldOperationInfo("sample_pipeline.stage1.read", "read",
-            "reading a file", new FieldOperationInput(Collections.singletonList(sourceEndPoint), null),
+    FieldOperationInfo operationInfo = new FieldOperationInfo("read", "reading a file",
+            new FieldOperationInput(Collections.singletonList(sourceEndPoint), null),
             new FieldOperationOutput(null, Collections.singletonList("body")));
     fieldOperationInfos.add(operationInfo);
-    operationInfo = new FieldOperationInfo("sample_pipeline.stage2.parse", "parse",
-            "parsing a comma separated field",
-            new FieldOperationInput(null,
-                    Collections.singletonList(InputField.of("sample_pipeline.stage1.read", "body"))),
+    operationInfo = new FieldOperationInfo("parse", "parsing a comma separated field",
+            new FieldOperationInput(null, Collections.singletonList(InputField.of("read", "body"))),
```

Comment:
would it make sense to have static constructor methods such as ofFields() and ofDataset(). Seems more intuitive than a constructor that takes two parameters, and only one of them can be non-null

Reply:
added static methods - `of(EndPoint endPoint` and `of(List<InputFields>)`. 
Realized that we only need to store single EndPoint for FieldOperationInput and FieldOperationOutput so fixed that too.